inference-engine/tests/functional/plugin/shared/include/single_layer_tests/gather_nd.hpp

```diff
@@ -0,0 +1,41 @@
+// Copyright (C) 2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <memory>
+#include <string>
+#include <tuple>
+#include <vector>
+
+#include "functional_test_utils/layer_test_utils.hpp"
+#include "ngraph_functions/builders.hpp"
+#include "ngraph_functions/utils/ngraph_helpers.hpp"
+
+namespace LayerTestsDefinitions {
+
+using gatherNDLayerParams = std::tuple<
+    std::size_t,                   // Batch dims
+    InferenceEngine::SizeVector,   // Input shape
+    InferenceEngine::SizeVector>;  // Indices shape
+
+using gatherNDParams = std::tuple<
+    gatherNDLayerParams,                  // Layer params
+    InferenceEngine::Precision,           // Input and output precision
+    InferenceEngine::Precision,           // Indices precision
+    InferenceEngine::Layout,              // Input layout
+    InferenceEngine::Layout,              // Output layout
+    LayerTestsUtils::TargetDevice,        // Device name
+    std::map<std::string, std::string>>;  // Plugin config
+
+class GatherNDLayerTest : public testing::WithParamInterface<gatherNDParams>,
+                          virtual public LayerTestsUtils::LayerTestsCommon {
```

Comment:
@iefode is virtual inheritance required? Is it possible that someone would need to inherit from `GatherNDLayerTest`  and `LayerTestsCommon` or its child?
I see that it is often applied. Could you help me figure it out?

Reply:
Virtual inheritance was implemented for KMB plugin

Reply:
@iefode thank you for the clarification!